src/Groups/group_constructors.jl

```diff
@@ -827,6 +827,8 @@ as an instance of `T`,
 where `n` is a power of 2 and `T` is in
```

Comment:
```suggestion
where `n` is a multiple of 4  and `T` is in
```

Reply:
I think "power of 2" is what we want here, as that is how (generalized) quaternion groups are defined. For the function that does the exact same construction for "multiple of 4", there is `dicyclic_group` below. In particular, this distinction in the docstring matches to how `is_quaternion_group` works

Reply:
No, the consensus in https://github.com/oscar-system/Oscar.jl/issues/1630 is that both should be the same.
1. The definition with 2-powers is unnecessarily restrictive. There are plenty of references that construct orders of group 4n.
2. There is now already code out there that is using the right definition (that the order is any number $4n$), because this is what people were being told in #1630
3. Since we don't do any quaternion_group(k) -> group of order 2^k or 2^(k+2) shenanigans, there is no reason to restrict to 2-powers.

Reply:
Yeah I decided to just treat both commands as exact equals, any artificial restriction benefits nobody but can cause frustration

Reply:
Note that e.g. [Wikipedia](https://en.wikipedia.org/wiki/Quaternion_group#Generalized_quaternion_group) allows "generalized quaternion groups" to have order $4k$.